### PR TITLE
Space Bar Holomenu + Bar-sign Fix

### LINKED ISSUE #13477: Space Bar Bartender has no access to the barsign or holo-menu
image

image
image

image
image

Asides from these two specific objects, you have access to anywhere else in the outpost, Booze-O-Mat included. Not role-breaking, but still something that should be fixed sometime.

### PR DESCRIPTION
**Pull request by Ben10083**
Fixes #13477 by having barsign entity check for access instead of assuming req_access is for just the bar.

### CODE REVIEW

**Comment by Ben10083** [comment]
!bugfix

File: maps/sccv_horizon/sccv_horizon-2_deck_2.dmm [hunk 1]
```diff
@@ -22257,7 +22257,8 @@
 	},
 /obj/structure/sign/double/barsign{
 	dir = 4;
-	pixel_x = -64
+	pixel_x = -64;
+	req_access = list(25)
```
Comment (NonQueueingMatt) [inline maps/sccv_horizon/sccv_horizon-2_deck_2.dmm:22261]: I don't think this should be defined? It's the Horizon's bar, so `access_bar` should be fine here.

Comment (Ben10083) [reply maps/sccv_horizon/sccv_horizon-2_deck_2.dmm:22261]: spacebar.dm assumes at first that access_bar is present, but later on checks req_access when it does (check_access(card)).

If this is not the case I will revert my change to deck 2

Comment (Ben10083) [reply maps/sccv_horizon/sccv_horizon-2_deck_2.dmm:22261]: image
Confirmed it, if I leave req_access null anyone can access it, this is me changing the sign as janitor.

Comment (JohnWildkins) [reply maps/sccv_horizon/sccv_horizon-2_deck_2.dmm:22261]: You need to delete it entirely, not set it to null
Having the `req_access` var mapped in on that sign either way overwrites the `req_access = list(access_bar)` in the object's definition, thus letting anyone access it when you set it to null
Any variables mapped in overwrite those set in code

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**